server/game/core/Player.js

```diff
@@ -919,21 +927,29 @@ class Player extends GameObject {
      */
     // TODO: Create an ExhaustOrReadyResourcesSystem
     exhaustResources(count, priorityResources = []) {
-        let priorityResourcesExhausted = 0;
-        let readyResources = this.resources.filter((card) => !card.exhausted);
-        if (priorityResources.length > 0) {
+        Contract.assertTrue(this.countSpendableResources() >= count,
```

Comment:
@Ennis0 was also working on this method and convinced me that this isn't necessary, we should be allowed to try and exhaust / ready more resources than available and it will just do as many as it can. What would your vote be?

Reply:
@Ennis0 @AMMayberry1 Would an example be like if you attack with Boba but can only ready 1 resource?

Reply:
Yeah correct. It's just a design philosophy question, it's more flexible if we let the caller ask for more resources than are available but could allow bugs to sneak through. Forcing the caller to check how many are available first is more work but might catch some issues occasionally. It's a small enough thing that I'm good either way.

Reply:
@AMMayberry1 I think it depends to me whether the ability is something we can choose. Like if we are not giving them the option of when to choose Boba's leader-side ability, then we should make sure it uses as much, but if we let them choose, we could probably get rid of this check? Does that make sense?